### PR TITLE
add test for hosted opensearch

### PR DESCRIPTION
**Pull request by niedbalski**
* Add terraform resources required for running hosted opensearch 1.2
* Add terraform resources for handling logs
* Adds linters (tflint/tfsec) to github workflows.

### CODE REVIEW

File: terraform/variables.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  default     = "us-east-1"
+}
+
+variable "aws_access_id" {
+  type        = string
+  description = "AWS access ID"
+}
+
+variable "aws_secret_key" {
+  type        = string
```
Comment (patrick-stephens) [inline terraform/variables.tf:13]: Probably need to mark some of these as sensitive

Comment (niedbalski) [reply terraform/variables.tf:14]: done

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -17,6 +17,42 @@ jobs:
...
+      - name: Terraform init
+        run: terraform init
+        working-directory: terraform/
+
+      - name: Terraform validate
+        run: terraform validate -no-color
+        working-directory: terraform/
+
+      - name: Terraform plan
+        if: ${{ github.event_name != 'pull_request' }}
```
Comment (patrick-stephens) [inline .github/workflows/ci.yaml:42]: Remove

File: tests/opensearch/hosted.bats [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    if [ -e  "${HELM_VALUES_EXTRA_FILE}" ]; then
+      envsubst < "${HELM_VALUES_EXTRA_FILE}" > "${HELM_VALUES_EXTRA_FILE%.*}"
+      export HELM_VALUES_EXTRA_FILE="${HELM_VALUES_EXTRA_FILE%.*}"
+    fi
+}
+
+teardown() {
+    if [[ "${SKIP_TEARDOWN:-no}" != "yes" ]]; then
+        run kubectl delete namespace "$TEST_NAMESPACE"
+        [ -e ${HELM_VALUES_EXTRA_FILE} ] && rm ${HELM_VALUES_EXTRA_FILE}
```
Comment (patrick-stephens) [inline tests/opensearch/hosted.bats:31]: `rm -f` to remove any confirmation plus no need for check (will not fail if does not exist)

File: tests/opensearch/hosted.bats [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAME="kubectl -n $TEST_NAMESPACE"
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAMESPACE="${TEST_NAMESPACE}"
+
+
+@test "test fluent-bit forwards logs to AWS OpenSearch hosted service default index" {
+    envsubst < "${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml.tpl" > "${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml"
+    cat ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml
+    helm upgrade --install --debug --namespace "$TEST_NAMESPACE" fluent-bit fluent/fluent-bit --values $HELM_VALUES_EXTRA_FILE -f ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml --set image.repository=${FLUENTBIT_IMAGE_REPOSITORY} --set image.tag=${FLUENTBIT_IMAGE_TAG} --wait
```
Comment (patrick-stephens) [inline tests/opensearch/hosted.bats:46]: Can you split into multiple lines so not all on one long one?

**Comment by niedbalski** [comment]
@patrick-stephens all addressed

File: tests/opensearch/hosted.bats [hunk 5]
```diff
@@ -42,8 +42,12 @@ DETIK_CLIENT_NAMESPACE="${TEST_NAMESPACE}"
 
 @test "test fluent-bit forwards logs to AWS OpenSearch hosted service default index" {
     envsubst < "${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml.tpl" > "${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml"
-    cat ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml
-    helm upgrade --install --debug --namespace "$TEST_NAMESPACE" fluent-bit fluent/fluent-bit --values $HELM_VALUES_EXTRA_FILE -f ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml --set image.repository=${FLUENTBIT_IMAGE_REPOSITORY} --set image.tag=${FLUENTBIT_IMAGE_TAG} --wait
+
+    helm upgrade --install --debug --namespace "$TEST_NAMESPACE" fluent-bit fluent/fluent-bit \
```
Comment (patrick-stephens) [inline tests/opensearch/hosted.bats:46]: `--create-namespace` required as well?

File: tests/opensearch/hosted.bats [hunk 6]
```diff
@@ -28,8 +28,8 @@ setup() {
 teardown() {
     if [[ "${SKIP_TEARDOWN:-no}" != "yes" ]]; then
         run kubectl delete namespace "$TEST_NAMESPACE"
-        [ -e ${HELM_VALUES_EXTRA_FILE} ] && rm ${HELM_VALUES_EXTRA_FILE}
-        [ -e ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml ] && rm ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-hosted.yaml
+        [ -e ${HELM_VALUES_EXTRA_FILE} ] && rm -f ${HELM_VALUES_EXTRA_FILE}
```
Comment (patrick-stephens) [inline tests/opensearch/hosted.bats:31]: No need to test for file existence

**Pull request merged**